internal/backends/java/java.go

```diff
@@ -0,0 +1,201 @@
+// Package java provides a backend for Java using maven.
+package java
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/replit/upm/internal/api"
+	"github.com/replit/upm/internal/util"
+)
+
+type Dependency struct {
+	XMLName    xml.Name `xml:"dependency"`
+	GroupId    string   `xml:"groupId"`
+	ArtifactId string   `xml:"artifactId"`
+	Version    string   `xml:"version"`
+}
+
+type Project struct {
+	XMLName      xml.Name     `xml:"project"`
+	ModelVersion string       `xml:"modelVersion"`
+	GroupId      string       `xml:"groupId"`
+	ArtifactId   string       `xml:"artifactId"`
+	Version      string       `xml:"version"`
+	Dependencies []Dependency `xml:"dependencies>dependency"`
+}
+
+var emptyProject = Project{
+	ModelVersion: "4.0.0",
+	GroupId:      "co.repl",
+	ArtifactId:   "artifact",
+	Version:      "0.0-SNAPSHOT",
+	Dependencies: []Dependency{},
+}
+
+var pkgNameRegexp = regexp.MustCompile("^([^:]+):([^:]+)") // groupid:artifactid
+
+// javaPatterns is the FilenamePatterns value for JavaBackend.
+var javaPatterns = []string{"*.java"}
+
+func readProjectOrMakeEmpty(path string) Project {
+	var project Project
+	if util.Exists("pom.xml") {
+		xmlbytes, err := ioutil.ReadFile("pom.xml")
+		if err != nil {
+			util.Die("error reading pom.xml: %s", err)
+		}
+		err = xml.Unmarshal(xmlbytes, &project)
+		if err != nil {
+			util.Die("error unmarshalling pom.xml: %s", err)
+		}
+	} else {
+		project = emptyProject
+	}
+	return project
+}
+
+const pomdotxml = "pom.xml"
+
+// JavaBackend is the UPM language backend for Java using Maven.
+var JavaBackend = api.LanguageBackend{
+	Name:             "java-maven",
+	Specfile:         pomdotxml,
+	Lockfile:         pomdotxml,
+	FilenamePatterns: javaPatterns,
+	Quirks:           api.QuirksAddRemoveAlsoLocks,
+	GetPackageDir: func() string {
+		return "target/dependency"
+	},
+	Search: func(query string) []api.PkgInfo {
+		searchDocs, err := Search(query)
+		if err != nil {
+			util.Log("error searching maven %s", err)
```

Comment:
what's the reason for logging but not exiting with an error?

Reply:
no reason! changing many of these to `util.Die()`